File: frontend/types/media.ts
// frontend/types/media.ts

export type MediaType = "image" | "video";
export type MediaStatus = "uploading" | "processing" | "ready" | "failed";

export type AiStatus = "none" | "running" | "ready" | "failed";

export interface MediaListItem {
  id: string;
  filename: string;
  media_type: MediaType;
  status: MediaStatus;
  created_at: string;
  thumbnail_url: string | null;

  // persisted analysis metadata (optional)
  analysis_type?: string | null;
  notes?: string | null;
  location?: string | null;
  capture_date?: string | null; // "YYYY-MM-DD"

  // ✅ NEW: ai status from backend
  ai_status?: AiStatus | null;
  latest_job_id?: number | null;
  latest_job_status?: string | null;
  latest_job_type?: string | null;
}

export interface MediaListResponse {
  items: MediaListItem[];
}

export interface MediaUploadIntentRequest {
  filename: string;
  media_type: MediaType;
  content_type?: string;
  size_bytes?: number;

  analysis_type?: string | null;
  notes?: string | null;
  location?: string | null;
  capture_date?: string | null;
}

export interface MediaUploadIntentResponse {
  media: MediaListItem;
  upload_url: string;
  upload_method: "PUT" | "POST";
  upload_headers?: Record<string, string>;
  upload_fields?: Record<string, string>;
}

File: backend/app/schemas/media.py
# app/schemas/media.py

from datetime import datetime, date
from typing import Literal, Dict, Optional
from pydantic import BaseModel

MediaType = Literal["image", "video"]
MediaStatus = Literal["uploading", "processing", "ready", "failed"]

AiStatus = Literal["none", "running", "ready", "failed"]


class MediaListItem(BaseModel):
    id: str
    filename: str
    media_type: MediaType
    status: MediaStatus
    created_at: datetime
    thumbnail_url: str | None = None

    # persisted analysis metadata
    analysis_type: Optional[str] = None
    notes: Optional[str] = None
    location: Optional[str] = None
    capture_date: Optional[date] = None

    # ✅ NEW: ai status fields for library consistency
    ai_status: Optional[AiStatus] = None
    latest_job_id: Optional[int] = None
    latest_job_status: Optional[str] = None
    latest_job_type: Optional[str] = None


class MediaListResponse(BaseModel):
    items: list[MediaListItem]


class MediaDetailResponse(MediaListItem):
    pass


class MediaUploadIntentRequest(BaseModel):
    filename: str
    media_type: MediaType
    content_type: Optional[str] = None
    size_bytes: Optional[int] = None

    # persisted analysis metadata
    analysis_type: Optional[str] = None
    notes: Optional[str] = None
    location: Optional[str] = None
    capture_date: Optional[date] = None


class MediaUploadIntentResponse(BaseModel):
    media: MediaListItem
    upload_url: str
    upload_method: Literal["PUT", "POST"]
    upload_headers: Optional[Dict[str, str]] = None
    upload_fields: Optional[Dict[str, str]] = None


class MediaFileUrlResponse(BaseModel):
    url: str

